src/redis-cli.c

```diff
@@ -4708,6 +4709,43 @@ static clusterManagerNode *clusterManagerGetSlotOwner(clusterManagerNode *n,
 static int clusterManagerSetSlot(clusterManagerNode *node1,
                                  clusterManagerNode *node2,
                                  int slot, const char *status, char **err) {
+    /* A new CLUSTER SETSLOT variant is introduced in Redis
+     * 7.2+ to help mitigate the single-point-of-failure
+     * issue related to the slot ownership finalization on
+     * HA clusters. We make a best-effort attempt below to
+     * utilize this enhanced reliability. Regardless of the
+     * result, we continue with finalizing slot ownership
+     * on the primary nodes. Note that this command is not
+     * essential. Redis 7.2+ will attempt to recover from
+     * failed slot ownership finalizations if they occur,
+     * although there may be a brief period where slots
+     * caught in this transition stage are unavailable.
+     * Including this additional step ensures no downtime
+     * for these slots if any failures arise. */
+    if (status != NULL && !strncmp(status, "node", 4)) {
+        /* Caller is attempting to finalize slot ownership.
+         * Inject the new command to attempt finalization
+         * on replicas first. */
+        redisReply *reply = CLUSTER_MANAGER_COMMAND(node1, "CLUSTER "
+                                                    "SETSLOT %d %s %s REPLICAONLY",
+                                                    slot, status,
+                                                    (char *) node2->name);
```

Comment:
This function `clusterManagerSetSlot` used to sent only one CLUSTER SETSLOT command. Are you sure we should "inject" the new command here and send two commands? Why not add the new command at the caller site? There are some calls to clusterManagerSetSlot with status="node" where we might not want to inject the new command, such as the calls from `clusterManagerFixSlotsCoverage()` and `clusterManagerFixOpenSlot()`.

Btw, `status` is always "node" here, so no need for %s.

Reply:
You are right. This change is too broad. We only need to perform the replication when finalizing slots in the target shard. Fixed.